packages/core/EventManager.ts

```diff
@@ -400,20 +404,32 @@ export default class EventManager {
             (c) => c.type === destination.integration
           );
           // It might not be the first connected calendar as it seems that the order is not guaranteed to be ascending of credentialId.
-          const firstCalendarCredential = destinationCalendarCredentials[0];
-          log.warn(
-            "No credentialId found for destination calendar, falling back to first found calendar",
-            safeStringify({
-              destination: getPiiFreeDestinationCalendar(destination),
-              firstConnectedCalendar: getPiiFreeCredential(firstCalendarCredential),
-            })
-          );
-
-          createdEvents.push(await createEvent(firstCalendarCredential, event));
+          const firstCalendarCredential: (typeof destinationCalendarCredentials)[number] | undefined =
```

Comment:
With our current configuration, TypeScript doesn't consider it possibly undefined which it could be.